sdk/src/main/scala/io/horizen/certificatesubmitter/strategies/WithKeyRotationCircuitStrategy.scala

```diff
@@ -25,11 +28,11 @@ class WithKeyRotationCircuitStrategy[
   PM <: SidechainBlockBase[TX, H],
   HIS <: AbstractHistory[TX, H, PM, _, _, _],
   MS <: AbstractState[TX, H, PM, MS]](settings: SidechainSettings,
-                                      sc2scConfig: Sc2ScConfigurator,
                                       params: NetworkParams,
-                                      circuit: ThresholdSignatureCircuitWithKeyRotation)
-  extends CircuitStrategy[TX, H, PM, HIS, MS, CertificateDataWithKeyRotation](settings, sc2scConfig, params) with Sc2ScUtils[TX, H, PM, MS, HIS] {
-
+                                      circuit: ThresholdSignatureCircuitWithKeyRotation,
+                                      timeProvider: TimeProvider)
+  extends CircuitStrategy[TX, H, PM, HIS, MS, CertificateDataWithKeyRotation](settings, params) with Sc2ScUtils[TX, H, PM, MS, HIS] {
+  val sc2ScFork: Sc2ScFork = Sc2ScFork.get(TimeToEpochUtils.timeStampToEpochNumber(params, timeProvider.time()))
```

Comment:
Are we safe using the time provider for calculating the fork? The fork is probably not active at the initialization of the strategy since it is done at the app startup `(AccountSidechainApp.scala`) when the `AccountCertificateSubmitterRef` is created.